test/rulesets/LinearAlgebra/symmetric.jl

```diff
@@ -283,4 +283,225 @@
             @test @inferred(back(Zero())) == (NO_FIELDS, Zero())
         end
     end
+
+    @testset "Symmetric/Hermitian matrix functions" begin
+        # generate random matrices of type TA in the domain of f
+        function rand_matfun_input(f, TA, T, uplo, n, hermout)
+            U = Matrix(qr(randn(T, n, n)).Q)
+            if hermout # f(A) will also be a TA
+                λ = if f in (acos, asin, atanh)
+                    2 .* rand(real(T), n) .- 1
+                elseif f in (log,sqrt)
+                    abs.(randn(real(T), n))
+                elseif f === acosh
+                    1 .+ abs.(randn(real(T), n))
+                else
+                    randn(real(T), n)
+                end
+            else
+                λ = randn(real(T), n)
+                λ = if f === atanh
+                    2 .* rand(real(T), n) .- 1
+                else
+                    randn(real(T), n)
+                end
+            end
+            return TA(U * Diagonal(λ) * U', uplo)
+        end
+
+        @testset "$(f)(::$TA{<:$T})" for f in
+            (exp, log, sqrt, cos, sin, tan, cosh, sinh, tanh, acos, asin, atan, acosh, asinh, atanh),
+            TA in (Symmetric, Hermitian),
+            T in (TA <: Symmetric ? (Float64,) : (Float64, ComplexF64))
+
+            n = 10
+            @testset "frule" begin
+                @testset for uplo in (:L, :U), hermout in (true, false)
+                    A, ΔA = rand_matfun_input(f, TA, T, uplo, n, hermout), TA(randn(T, n, n), uplo)
+                    Y = f(A)
+                    istypestable = try
+                        @inferred f(A)
+                        true
+                    catch ErrorException
+                        false
+                    end
```

Comment:
can we extract this out into a helper function?

Reply:
Yeah, I can do that. Actually, we should have this in ChainRulesTestUtils anyways, since it doesn't make sense to check if an `frule` or `rrule` is type-stable if its primal isn't (its pullback can still be though).

Reply:
I added the helper function to https://github.com/JuliaDiff/ChainRulesTestUtils.jl/pull/103 and will use it in this PR.